mgmtd/mgmt_txn.c

```diff
@@ -2820,22 +2823,32 @@ int mgmt_txn_notify_rpc_reply(struct mgmt_be_client_adapter *adapter,
 
 	rpc = txn_req->req.rpc;
 
-	/* we don't expect more than one daemon to provide output for an RPC */
-	if (!rpc->client_results && data_len > 0) {
+	tree = NULL;
+	if (data_len)
 		err = yang_parse_rpc(rpc->xpath, reply_msg->result_type,
-				     reply_msg->data, true,
-				     &rpc->client_results);
+				     reply_msg->data, true, &tree);
+	if (err) {
+		__log_err("RPC reply from %s for txn-id %" PRIu64
+			  " req_id %" PRIu64 " error parsing result of type %u",
+			  adapter->name, txn_id, req_id, reply_msg->result_type);
```

Comment:
if this is a log message (rather than a debug), is there a string form of the error that's available?

Reply:
There is actually (it's an LY_ERR and there's an err2str function), i'll add it to the 2 messages.